Cabal/Distribution/Backpack/UnifyM.hs

```diff
@@ -318,7 +391,9 @@ convertUnitIdU' stk uid_u = do
         UnitIdThunkU uid -> return (DefiniteUnitId uid)
         UnitIdU u cid insts_u ->
             case lookupMooEnv stk u of
-                Just _i -> error "convertUnitIdU': mutual recursion" -- return (UnitIdVar i)
+                Just _i ->
+                    failWith (text "Unsupported mutually recursive unit identifier")
+                    -- return (UnitIdVar i)
```

Comment:
Well, it's what you would replace the line with if you added support for mutual recursion :)